src/views/Combine.vue

```diff
@@ -53,69 +53,82 @@
   </div>
 </template>
 
-<script>
+<script lang="ts">
 /*eslint no-console: ["error", { allow: ["warn", "error"] }] */
 
-import crypto from "../util/crypto";
+import crypto, { Shard } from "../util/crypto";
+import Vue from "vue";
 
-export default {
+type CombineData = {
+  title: string;
+  nonce: string;
+  shards: Shard[];
+  qrCodes: Set<string>;
+  requiredShards?: number;
+  passphrase: string;
+  recoveredSecret?: string;
+  PLACEHOLDER_QR_DATA: string;
+};
+
+export default Vue.extend({
   name: "Combine",
-  data: function() {
+  data(): CombineData {
     return {
       title: "",
       nonce: "",
-      shards: new Set(),
```

Comment:
It's correct?

Reply:
Yes, I slightly changed the logic: now `qrCodes` is a Set and `shards` is an array.